### ISSUE TITLE
gs: failed to unlock correct collection '/org/freedesktop/secrets/collection/login'

### ISSUE THREAD

**Issue by messense** [abhinav/git-spice#293]
```bash
$ gs auth login
Select an authentication method: GitHub CLI
FTL gs: failed to unlock correct collection '/org/freedesktop/secrets/collection/login'
```

I'm on Ubuntu 22.04 server headless environment.

BTW, does git-spice support GitHub Enterprise Server?

**Comment by abhinav** [comment]
Hello!

On Linux, git-spice uses [Secret Service](https://specifications.freedesktop.org/secret-service/latest/) for secure token storage. GNOME Keyring is one of the providers for Secret Service; you probably can't install it in a headless environment.
If you don't have a means of secure storage, git-spice could add support for a non-secure fallback storage.

Although for GitHub CLI specifically, we could probably get away without using any secret storage. I'll look into what we can do here in the coming days.

If you would like an immediate workaround, set the GITHUB_TOKEN environment variable and you won't need to authenticate at all.

> BTW, does git-spice support GitHub Enterprise Server?

While I don't have the means of testing against GHES, if the API is the same, it should work.
Set the GITHUB_URL and GITHUB_API_URL environment variables to hit a destination other than github.com.
(I'll add that to the docs if anyone using GHES can verify that this is enough.)

**Comment by messense** [comment]
> If you don't have a means of secure storage, git-spice could add support for a non-secure fallback storage.

That'd be great, I don't have any secure storage, gnome-keyring does not seem to work for me without GUI.


> While I don't have the means of testing against GHES, if the API is the same, it should work.
> Set the GITHUB_URL and GITHUB_API_URL environment variables to hit a destination other than github.com.
> (I'll add that to the docs if anyone using GHES can verify that this is enough.)

Thanks, will try.

**Comment by messense** [comment]
> While I don't have the means of testing against GHES, if the API is the same, it should work.

I just tried on a GHES instance, it works when setting `GITHUB_URL`, `GITHUB_API_URL` and `GITHUB_TOKEN`. For example, suppose the GHES URL is `https://github.example.com`, then you need to set

```bash
export GITHUB_URL=https://github.example.com
export GITHUB_API_URL="$GITHUB_URL/api"
export GITHUB_TOKEN=<a valid token>
```

**Comment by abhinav** [comment]
Awesome, thanks for confirming! I'll add that to the documentation.

**Comment by abhinav** [comment]
Fixed in #298!
Thanks for reporting and helping here, @messense @klnusbaum.
I'll try to release this tomorrow. Feel free to try it out sooner by building from source.
